tests/src/test/java/org/eclipse/hono/tests/http/HttpTestBase.java

```diff
@@ -911,6 +911,36 @@ public void testUploadMessageFailsForUnauthorizedGateway(final VertxTestContext
             .onComplete(ctx.succeedingThenComplete());
     }
 
+    /**
+     * Verifies that the HTTP adapter rejects messages with content length > max body limit size.
+     *
+     * @param ctx The test context
+     */
+    @Test
+    @Timeout(timeUnit = TimeUnit.SECONDS, value = 20)
+    public void testUploadMessageFailsForBodyLimitSize(final VertxTestContext ctx) {
+        final Tenant tenant = new Tenant();
+        final Device device = new Device();
+
+        helper.registry
+                .addDeviceForTenant(tenantId, tenant, deviceId, device, PWD)
+                .compose(ok -> {
+
+                    // WHEN the device tries to upload a message
+                    final MultiMap requestHeaders = MultiMap.caseInsensitiveMultiMap()
+                            .add(HttpHeaders.CONTENT_TYPE, "text/plain")
+                            .add(HttpHeaders.AUTHORIZATION, authorization);
+
+                    return httpClient.create(
+                            getEndpointUri(),
+                            Buffer.buffer(IntegrationTestSupport.getPayload(4096)),
+                            requestHeaders,
+                            ResponsePredicate.status(HttpURLConnection.HTTP_ENTITY_TOO_LARGE));
+                })
+                // THEN the message gets rejected with a 413
+                .onComplete(ctx.succeedingThenComplete());
```

Comment:
Assertions are missing here to verify the response status (see bottom of `testHandleConcurrentUploadWithTtd` for example on how to verify and complete the context in such a case).